src/ts/BusinessLogicIndex.ts

```diff
@@ -0,0 +1,94 @@
+
+import { Products } from "./interfaces/Products.js"
+import { Inventory } from "./Inventory.js"
+import { Storage } from "./Storage.js"
+
+export class BusinessLogic extends Storage<Products>{
+
+    getProducts(section: HTMLElement, index: number): void {
+        let initialPoint = 0
+        let finalPoint = 0
+        if (index === 0) {
+            initialPoint = 0
+            finalPoint = 4
+        }
+        if (index === 1) {
+            initialPoint = 4
+            finalPoint = 8
+        }
+        section.innerHTML = this.template(initialPoint, finalPoint)
```

Comment:
Ao invés de redefinir a variável, poderia chamar direto o `this.template()` com os valores corretos dentro do `if` e encerrar a execução do método.